src/std/io/bio/bio-test.ss

```diff
@@ -0,0 +1,250 @@
+        (BufferedWriter-close bwr)
+        (check (get-buffer-output-u8vector bwr) => u8v)))
+
+    (test-case "integer output"
+      (let* ((u8v (u8vector
+                   ;; u16
+                   #x01 #x02
+                   ;; u32
+                   #x01 #x02 #x03 #x04
+                   ;; u64
+                   #x01 #x02 #x03 #x04 #x05 #x06 #x07 #x08
+                   ;; s16
+                   #x01 #x02
+                   #x81 #x02
+                   ;; s32
+                   #x01 #x02 #x03 #x04
+                   #x81 #x02 #x03 #x04
+                   ;; s64
+                   #x01 #x02 #x03 #x04 #x05 #x06 #x07 #x08
+                   #x81 #x02 #x03 #x04 #x05 #x06 #x07 #x08
+                   ;; varuint
+                   #xaf #x96 #x13
+                   ;; varint
+                   #xde #xac #x26
+                   #xdd #xac #x26
+                   ))
+             (bwr (open-buffered-writer #f)))
+        (check (BufferedWriter-write-u16 bwr #x0102) => 2)
+        (check (BufferedWriter-write-u32 bwr #x01020304) => 4)
+        (check (BufferedWriter-write-u64 bwr #x0102030405060708) => 8)
+        (check (BufferedWriter-write-s16 bwr #x0102) => 2)
+        (check (BufferedWriter-write-s16 bwr -33027) => 2)
+        (check (BufferedWriter-write-s32 bwr #x01020304) => 4)
+        (check (BufferedWriter-write-s32 bwr -2164392709) => 4)
+        (check (BufferedWriter-write-s64 bwr #x0102030405060708) => 8)
+        (check (BufferedWriter-write-s64 bwr -9295995896645158665) => 8)
+        (check (BufferedWriter-write-varuint bwr 314159) => 3)
+        (check (BufferedWriter-write-varint bwr 314159) => 3)
+        (check (BufferedWriter-write-varint bwr -314159) => 3)
+        (BufferedWriter-close bwr)
+        (check (get-buffer-output-u8vector bwr) => u8v)))
+
+    (test-case "char output"
+      (let* ((input "the quick brown fox jumped over the fence")
+             (output (string->utf8 input))
+             (bwr (open-buffered-writer #f)))
+        (for (char (string->list input))
+          (check (BufferedWriter-write-char bwr char) => 1))
+        (BufferedWriter-close bwr)
+        (check (get-buffer-output-u8vector bwr) => output)))
+
+    (test-case "string output"
+      (let* ((input "the quick brown fox jumped over the fence")
+             (output (string->utf8 input))
+             (bwr (open-buffered-writer #f)))
+        (for (i (in-range (fx/ (fx+ (string-length input) 15) 16)))
+          (let* ((input-start (* i 16))
+                 (input-end (fxmin (* (+ i 1) 16) (string-length input)))
+                 (expected-chars (fx- input-end input-start)))
+            (check (BufferedWriter-write-string bwr input input-start input-end) => expected-chars)))
+        (BufferedWriter-close bwr)
+        (check (get-buffer-output-u8vector bwr) => output)))
+
+    (test-case "line output"
+      (let ((input "the quick brown fox jumped over the fence")
```

Comment:
Why the fence? With the lazy dog, it's a pangram. Or use other pangrams:
```
"Mr. Jock, TV quiz PhD, bags few lynx."
"Squdgy fez, blank jimp crwth vox!"
"Waltz, nymph, for quick jigs vex Bud."
"Sphinx of black quartz, judge my vow."
"Pack my box with five dozen liquor jugs"
```

Reply:
I'll make our foxes jump over lazy dogs.

Reply:
done.